test/browser/spec/ol/renderer/map.test.js

```diff
@@ -205,6 +205,174 @@ describe('ol.renderer.Map', function () {
+        feature,
+        layer,
+        geometry,
+      }));
+      expect(hit).to.be.ok();
+      expect(hit.feature).to.be(feature);
+      expect(hit.layer).to.be(layer);
+      expect(hit.geometry).to.be(geometry);
+
+      layer.setStyle({
+        'text-value': 'X',
+        'text-font': 'bold 100px sans-serif',
+        'text-baseline': 'top',
+        'text-offset-y': -50,
+        'text-stroke-width': 1,
+        'text-stroke-color': 'black',
+        'text-fill-color': null,
+      });
+      map.renderSync();
+      hit = map.forEachFeatureAtPixel([50, 50], (feature, layer, geometry) => ({
+        feature,
+        layer,
+        geometry,
+      }));
+      expect(hit).to.be(undefined);
+
+      layer.setStyle({
+        'text-value': 'X',
+        'text-font': 'bold 100px sans-serif',
+        'text-baseline': 'top',
+        'text-offset-y': -50,
+        'text-stroke-width': 1,
+        'text-stroke-color': 'black',
+        'text-fill-color': 'transparent',
+      });
+      map.renderSync();
+      hit = map.forEachFeatureAtPixel([50, 50], (feature, layer, geometry) => ({
+        feature,
+        layer,
+        geometry,
+      }));
+      expect(hit).to.be.ok();
+      expect(hit.feature).to.be(feature);
+      expect(hit.layer).to.be(layer);
+      expect(hit.geometry).to.be(geometry);
+
+      layer.setStyle({
+        'text-value': 'X',
+        'text-font': 'bold 100px sans-serif',
+        'text-baseline': 'top',
+        'text-offset-y': -50,
+        'text-stroke-width': 1,
+        'text-stroke-color': 'black',
+        'text-fill-color': null,
+        'text-background-fill-color': 'transparent',
+      });
+      map.renderSync();
+      hit = map.forEachFeatureAtPixel([50, 50], (feature, layer, geometry) => ({
+        feature,
+        layer,
+        geometry,
+      }));
+      expect(hit).to.be.ok();
+      expect(hit.feature).to.be(feature);
+      expect(hit.layer).to.be(layer);
+      expect(hit.geometry).to.be(geometry);
+    });
+
+    it('hits line placement Text stroke and transparent fill', function () {
+      let hit;
+      const geometry = new LineString([
+        [-1e6, 0],
+        [1e6, 0],
+      ]);
+      const feature = new Feature(geometry);
+      const layer = new VectorLayer({
+        source: new VectorSource({
+          features: [feature],
+        }),
+      });
+      map.addLayer(layer);
+
+      layer.setStyle({
+        'text-value': 'X',
+        'text-font': 'bold 100px sans-serif',
+        'text-baseline': 'top',
+        'text-offset-y': -50,
+        'text-stroke-width': 20,
+        'text-stroke-color': 'black',
+        'text-fill-color': null,
+        'text-placement': 'line',
+        'text-overflow': true,
+      });
+      map.renderSync();
+      hit = map.forEachFeatureAtPixel([50, 50], (feature, layer, geometry) => ({
+        feature,
+        layer,
+        geometry,
+      }));
+      expect(hit).to.be.ok();
```

Comment:
@mike-000 This keeps failing for me locally. Any chance you could look into this again please?

Reply:
There no reason why pixel [50, 50] would not hit detect for me 
![image](https://github.com/openlayers/openlayers/assets/49240900/327da460-1278-49d2-847b-9aabf4eaf199)
(taken from https://codesandbox.io/s/simple-forked-p7mmhh?file=/main.js)  I do see a small difference in the offset of the character if remove  `'text-placement': 'line',` but the font is so big that should not break it (and I presume the point feature test worked for you?).  If fonts differ between systems you might need a wider stroke to reach the center of the character.